File: solutions/Python/70.ClimbingStairs.py
class Solution:
    def climbStairs(self, n: int) -> int:
        if n <= 2:
            return n

        step1 = 1
        step2 = 2

        # When using range in python, range does not include n. So since n = 5, it goes up to 4 without including 5.
        for i in range(3, n + 1):
            temp = step2
            step2 += step1
            step1 = temp

        return step2

File: solutions/TypeScript/70.ClimbingStairs.ts
function climbStairs(n: number): number {
  if (n <= 2) return n;

  let step1: number = 1;
  let step2: number = 2;

  for (let i: number = 3; i <= n; i++) {
    let temp: number = step2;
    step2 += step1;
    step1 = temp;
  }

  return step2;
}
